.gitmodules

```diff
@@ -0,0 +1,4 @@
+[submodule "modules/mavlink"]
+	path = modules/mavlink
+	url = https://github.com/mavlink/mavlink.git
```

Comment:
Since you are going to add messages, you may want to have your own temporary fork of mavlink (to be reverted to upstream later).